packages/sd-jwt-vc/src/sd-jwt-vc-instance.ts

```diff
@@ -166,18 +166,31 @@ export class SDJwtVcInstance extends SDJwtInstance<SdJwtVcPayload> {
    * @param url
    * @returns
    */
-  private async fetch<T>(url: string, integrity?: string) {
+  private async fetch<T>(url: string, integrity?: string): Promise<T> {
     const controller = new AbortController();
-    const timeoutId = setTimeout(() => controller.abort(), 10000);
+    const timeoutId = setTimeout(
+      () => controller.abort(),
+      this.userConfig.timeout ?? 10000,
+    );
+
     try {
       const response = await fetch(url, {
         signal: controller.signal,
       });
+
       if (!response.ok) {
-        throw new Error(await response.text());
+        const errorText = await response.text();
+        throw new Error(
+          `Error fetching ${url}: ${response.status} ${response.statusText} - ${errorText}`,
+        );
       }
       await this.validateIntegrity(response.clone(), url, integrity);
       return response.json() as Promise<T>;
+    } catch (error) {
+      if (error instanceof DOMException && error.name === 'AbortError') {
```

Comment:
In node.js or react native environments, is this still a `DOMException`?

Reply:
@berendsliedrecht @lukasjhan I think we should write a test and find it out. I added this instance of check, otherwise the linter would fail because of the missing type.

I will open a new PR to add tests and to correct it

Reply:
@berendsliedrecht I ran the test in node `pnpm run test:node` and it threw me the DomException. I am not sure how it will work with react native since we have no chance here to test it...

One solution would be to replace fetch via a http library like axios. But this will add another external lib